From 36b64670fae87abd190f2896de24277c6c8ae8ac Mon Sep 17 00:00:00 2001
From: Janani Ravichandran <janani.rvchndrn@gmail.com>
Date: Thu, 25 Feb 2016 14:46:11 -0500
Subject: [PATCH 1/4] staging: fsl-mc: Drop unneeded void pointer cast
Patch-mainline: v4.6-rc1
Git-commit: 36b64670fae87abd190f2896de24277c6c8ae8ac
References: fate#319900

Void pointers need not be cast to other pointer types.
Semantic patch used:

@r@
expression x;
void *e;
type T;
identifier f;
@@

(
  *((T *)e)
|
  ((T *)x) [...]
|
  ((T *)x)->f
|
- (T *)
  e
)

Signed-off-by: Janani Ravichandran <janani.rvchndrn@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 drivers/staging/fsl-mc/bus/dprc-driver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/fsl-mc/bus/dprc-driver.c b/drivers/staging/fsl-mc/bus/dprc-driver.c
index 9380ebc..31488a7 100644
--- a/drivers/staging/fsl-mc/bus/dprc-driver.c
+++ b/drivers/staging/fsl-mc/bus/dprc-driver.c
@@ -407,7 +407,7 @@ static irqreturn_t dprc_irq0_handler_thread(int irq_num, void *arg)
 {
 	int error;
 	u32 status;
-	struct device *dev = (struct device *)arg;
+	struct device *dev = arg;
 	struct fsl_mc_device *mc_dev = to_fsl_mc_device(dev);
 	struct fsl_mc_bus *mc_bus = to_fsl_mc_bus(mc_dev);
 	struct fsl_mc_io *mc_io = mc_dev->mc_io;
-- 
2.6.2

